torch/csrc/distributed/c10d/CUDASymmetricMemory-inl.h

```diff
@@ -57,16 +69,16 @@ cas(uint32_t* addr, uint32_t compare, uint32_t val) {
 
 __device__ __forceinline__ void trap() {
 #if defined(USE_ROCM)
-  assert(0);
+  abort();
 #else
   __trap();
 #endif
 }
 
 __device__ __forceinline__ size_t global_timer_ns() {
 #if defined(USE_ROCM)
-  CUDA_KERNEL_ASSERT(false);
-  return 0;
+  static constexpr double MI300_FREQ_GHZ = 2.1;
```

Comment:
hmm, does this mean we'll have different number for MI350 (and also, we don't necessarily run @ 2.1GHz which is the max clock?

Reply:
Yeah, there isn't a way to grab timer on the GPU for ROCm. It's just the clock ticks. I need to divide that by the frequency, which would need to be read on the host side and passed to the kernel and that'd require a bunch of API changes. So, for now, I thought we can just carry a const until we have a better solution.